contracts/RelayEncoder.sol

```diff
@@ -0,0 +1,128 @@
+        return bytes.concat(hex"0613", scaleCompactUint(amount));
+    }
+
+    // dev Encode 'withdraw_unbonded' relay call
+    // @param slashes: Weight hint, number of slashing spans
+    // @returns The bytes associated with the encoded call
+    function encode_withdraw_unbonded(uint32 slashes) external pure returns (bytes memory) {
+        if (slashes < 1<<8) {
+            return bytes.concat(hex"0603", bytes1(uint8(slashes)), bytes3(0));
+        }
+        if(slashes < 1 << 16) {
+            uint32 bt2 = slashes / 256;
+            uint32 bt1 = slashes - bt2 * 256;
+            return bytes.concat(hex"0603", bytes1(uint8(bt1)), bytes1(uint8(bt2)), bytes2(0));
+        }
+        if(slashes < 1 << 24) {
+            uint32 bt3 = slashes / 65536;
+            uint32 bt2 = (slashes - bt3 * 65536) / 256;
+            uint32 bt1 = slashes - bt3 * 65536 - bt2 * 256;
+            return bytes.concat(hex"0603", bytes1(uint8(bt1)), bytes1(uint8(bt2)), bytes1(uint8(bt3)), bytes1(0));
+        }
+        uint32 bt4 = slashes / 16777216;
+        uint32 bt3 = (slashes - bt4 * 16777216) / 65536;
+        uint32 bt2 = (slashes - bt4 * 16777216 - bt3 * 65536) / 256;
+        uint32 bt1 = slashes - bt4 * 16777216 - bt3 * 65536 - bt2 * 256;
+        return bytes.concat(hex"0603", bytes1(uint8(bt1)), bytes1(uint8(bt2)), bytes1(uint8(bt3)), bytes1(uint8(bt4)));
+    }
+
+    // dev Encode 'nominate' relay call
+    // @param nominees: An array of AccountIds corresponding to the accounts we will nominate
+    // @param blocked: Whether or not the validator is accepting more nominations
+    // @returns The bytes associated with the encoded call
+    function encode_nominate(uint256 [] memory nominees) external pure returns (bytes memory) {
+        if (nominees.length == 0) {
+            return hex"060500";
+        }
+        bytes memory result = bytes.concat(hex"0605", scaleCompactUint(nominees.length));
+        for (uint256 i = 0; i < nominees.length; ++i) {
+            result = bytes.concat(result, hex"00", bytes32(nominees[i]));
+        }
+        return result;
+    }
+
+    // dev Encode 'chill' relay call
+    // @returns The bytes associated with the encoded call
+    function encode_chill() external pure returns (bytes memory) {
+        return hex"0606";
+    }
+
+    /**
+    * @notice Converting uint256 value to le bytes
+    * @param value - uint256 value
+    * @param len - length of output bytes array
+    */
+    function toLeBytes(uint256 value, uint256 len) internal pure returns(bytes memory) {
+        bytes memory out = new bytes(len);
+        for (uint256 idx = 0; idx < len; ++idx) {
+            out[idx] = bytes1(uint8(value));
+            value = value >> 8;
+        }
+        return out;
+    }
+
+    /**
+    * @notice Converting uint256 value to bytes
+    * @param value - uint256 value
+    */
+    function scaleCompactUint(uint256 value) internal pure returns(bytes memory) {
```

Comment:
I think we can move that piece of code to lib

Reply:
Done